prboom2/src/SDL/i_system.c

```diff
@@ -344,22 +344,38 @@ const char* I_GetTempDir(void)
 // cph - V.Aguilar (5/30/99) suggested return ~/.lxdoom/, creating
 //  if non-existant
 // cph 2006/07/23 - give prboom+ its own dir
-static const char prboom_dir[] = {"/.prboom-plus"}; // Mead rem extra slash 8/21/03
+static const char prboom_dir[] = {"prboom-plus"};
 
 const char *I_DoomExeDir(void)
 {
   static char *base;
+  struct stat data_dir;
+
   if (!base)        // cache multiple requests
     {
       char *home = getenv("HOME");
       size_t len = strlen(home);
+      size_t p_len = (len + strlen(prboom_dir) + 3);
 
-      base = malloc(len + strlen(prboom_dir) + 1);
-      strcpy(base, home);
       // I've had trouble with trailing slashes before...
-      if (base[len-1] == '/') base[len-1] = 0;
-      strcat(base, prboom_dir);
-      mkdir(base, S_IRUSR | S_IWUSR | S_IXUSR); // Make sure it exists
+      if (home[len-1] == '/') home[len-1] = 0;
+
+      base = malloc(p_len);
+      snprintf(base, p_len, "%s/.%s", home, prboom_dir);
+
+      stat(base, &data_dir);
+      if (!S_ISDIR(data_dir.st_mode))
+        {
+          char *prefpath = SDL_GetPrefPath("", prboom_dir);
+          size_t prefsize = strlen(prefpath);
+
+          free(base);
+          base = malloc(prefsize);
+          strcpy(base, prefpath);
+          // SDL_GetPrefPath always returns with trailing slash
+          if (base[prefsize-1] == '/') base[prefsize-1] = 0;
+          SDL_free(prefpath);
+        }
```

Comment:
Please keep the `mkdir()` part, the directory returnerd by `SDL_GetPrefPath()` does not necessarily exist already.

Reply:
SDL Wiki:
> This function will decide the appropriate location in the native filesystem, create the directory if necessary, and return a string of the absolute path to the directory in UTF-8 encoding.

I don't know if calling another `mkdir()` would change anything, but it's better to have redundancy than not.

Reply:
Alright, I didn't know about the "create the directory if necessary" part.

Reply:
I put it in as a comment, just in case SDL decides to switch things up.